sdk/storage/azure-storage-blob/azure/storage/blob/_download.py

```diff
@@ -433,6 +434,8 @@ def _get_encryption_data_request(self) -> None:
 
         # Restore cls for download
         self._request_options['cls'] = download_cls
+        if decompress:
```

Comment:
We should always put `True` or `False` back into the request options, this will leave it out if it is False. If we add the default to the above `pop`, this should change to `if decompress is not None` I think.

Reply:
Yup yup, please take this feedback when you apply this in your PR @weirongw23-msft